src/main/scala/ai/privado/languageEngine/csharp/processor/CSharpProcessor.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package ai.privado.languageEngine.csharp.processor
+
+import ai.privado.audit.AuditReportEntryPoint
+import ai.privado.cache.{AppCache, AuditCache, DataFlowCache, RuleCache, S3DatabaseDetailsCache, TaggerCache}
+import ai.privado.entrypoint.{ScanProcessor, TimeMetric}
+import ai.privado.exporter.{ExcelExporter, JSONExporter}
+import ai.privado.metric.MetricHandler
+import ai.privado.model.{CatLevelOne, Constants}
+import ai.privado.model.Constants.outputFileName
+import io.joern.dataflowengineoss.language.Path
+import io.joern.dataflowengineoss.layers.dataflows.{OssDataFlow, OssDataFlowOptions}
+import io.joern.x2cpg.X2Cpg
+import io.shiftleft.codepropertygraph
+import io.shiftleft.semanticcpg.layers.LayerCreatorContext
+import org.slf4j.LoggerFactory
+import ai.privado.model.Constants.*
+import ai.privado.utility.Utilities.createCpgFolder
+import better.files.File
+import io.joern.csharpsrc2cpg.*
+import io.shiftleft.semanticcpg.language.*
+import ai.privado.languageEngine.csharp.semantic.Language.tagger
+import ai.privado.dataflow.Dataflow
+import ai.privado.semantic.Language.*
+
+import java.util.Calendar
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try}
+
+object CSharpProcessor {
```

Comment:
@karan-batavia This style of implementation has been deprecated, Refer `Java` or `Kotlin` or `PHP`processor implementation